internal/repository/migrations/000001_create_application_input_claim_output_report_nodeconfig.up.sql

```diff
@@ -15,7 +15,7 @@ CREATE TABLE "application"
     "contract_address" BYTEA NOT NULL,
     "template_hash" BYTEA NOT NULL,
     "snapshot_uri" VARCHAR(4096) NOT NULL,
-    "last_processed_block" NUMERIC(20,0) NOT NULL,
+    "last_processed_block" NUMERIC(20,0) NOT NULL CHECK ("last_processed_block" >= 0 AND "last_processed_block" <= 18446744073709551615),
```

Comment:
I'm wondering if we could create a constant for `18446744073709551615` and reuses throughout the code.

Reply:
I checked if this was possible before doing it this way, but postgres has only three types of constants, none of which would support this large number https://www.postgresql.org/docs/7.0/syntax543.htm

To implement a constant like solution, we would need to use a function, which makes things more complicated, imho https://stackoverflow.com/questions/13316773/is-there-a-way-to-define-a-named-constant-in-a-postgresql-query

Reply:
I tried this 
```
CREATE FUNCTION public.f_maxuint64()
  RETURNS NUMERIC(20,0)
  LANGUAGE sql IMMUTABLE PARALLEL SAFE AS
'SELECT 18446744073709551615';


CREATE TABLE "application"
(
    "id" SERIAL,
    "contract_address" BYTEA NOT NULL,
    "template_hash" BYTEA NOT NULL,
    "snapshot_uri" VARCHAR(4096) NOT NULL,
    "last_processed_block" NUMERIC(20,0) NOT NULL CHECK ("last_processed_block" >= 0 AND "last_processed_block" <= f_maxuint64() ),
    "status" "ApplicationStatus" NOT NULL,
    "epoch_length" INT NOT NULL,
    CONSTRAINT "application_pkey" PRIMARY KEY ("id"),
    UNIQUE("contract_address")
);
```
And seems to work

Reply:
I was wondering if a function like this one would be optimized by postgres and what would be its effect in terms of performance. I guess [this explanation from the postgres docs](https://www.postgresql.org/docs/current/xfunc-optimization.html)is good enough, though.